python/taichi/lang/matrix_ops.py

```diff
@@ -0,0 +1,48 @@
+from taichi.lang.impl import static
+from taichi.lang.kernel_impl import func
+from taichi.lang.matrix_ops_utils import (is_tensor, preconditions,
+                                          square_matrix)
+from taichi.lang.misc import loop_config
+from taichi.lang.ops import cast
+from taichi.lang.util import taichi_scope
+
+
+@preconditions(square_matrix)
+@func
+def trace(x):
+    shape = static(x.get_shape())
+    result = cast(0, x.element_type())
+    # TODO: fix parallelization
+    loop_config(serialize=True)
+    for i in range(shape[0]):
```

Comment:
```suggestion
    for i in static(range(shape[0])):
```
Let's match the old implementation first :-)